src/superqt/utils/_throttler.py

```diff
@@ -227,16 +230,46 @@ def __call__(self, *args: P.args, **kwargs: P.kwargs) -> "Future[R]":  # noqa
         return self._future
 
     def _set_future_result(self):
-        result = self.__wrapped__(*self._args[: self._max_args], **self._kwargs)
+        if self._instance is not None:
+            result = self.__wrapped__(
+                self._instance, *self._args[: self._max_args], **self._kwargs
```

Comment:
One more q:
Did you double check that max_args doesn't need to include instance here?  (I.e, do we not have an off by one issue?)

Reply:
Based on this test, the self should not be count: 
https://github.com/pyapp-kit/superqt/blob/504adf8bd0b8302b11f252222dddff099cc8eb4e/tests/test_utils.py#L80-L93

Reply:
yeah, but max_args here is called on the unbound method (and that test is being called on a bound) method.  
It actually does fail:

```python
from qtpy.QtCore import QObject, Signal
from qtpy.QtWidgets import QApplication

from superqt.utils import qdebounced


class A(QObject):
    def __init__(self):
        super().__init__()
        self.count = 0

    @qdebounced(timeout=4)
    def callback(self, a):
        print(locals())


class E(QObject):
    signal = Signal(int, int)


app = QApplication([])
e = E()
a = A()
e.signal.connect(a.callback)
e.signal.emit(1, 2)
app.processEvents()

```

```
Traceback (most recent call last):
  File "/Users/talley/dev/self/superqt/src/superqt/utils/_throttler.py", line 234, in _set_future_result
    result = self.__wrapped__(
             ^^^^^^^^^^^^^^^^^
TypeError: A.callback() takes 2 positional arguments but 3 were given
zsh: abort      python x.py
```

I think you need to add 
```python
    def _set_future_result(self):
        if self._instance is not None:
            result = self.__wrapped__(
                *(self._instance, *self._args)[: self._max_args], **self._kwargs
            )
```

Reply:
Could you check my fix? It also handles staticmethod.

Reply:
also, looks like they changed staticmethod since py3.9... i think it might be more correct to swap the order of the decorators anyway?